gourmet/gtk_extras/TextBufferMarkup.py

```diff
@@ -78,26 +80,35 @@ def __init__ (self):
         GObject.GObject.__init__(self)
 
     def set_text (self, txt):
+        if isinstance(txt, bytes):
+            # data loaded from the database are bytes, not str
+            txt = txt.decode()
         Gtk.TextBuffer.set_text(self,"")
         try:
```

Comment:
while we're at it, I'd like your opinion @maweki: I'd like to remove the `try/except` block, by pre-emptively calling `saxutils.escape`.

The block:
```python
try:
    self.parsed, attributes, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')
except Exception as e:  # Pango getting unescaped txt text (g-markup-error-quark), eg. contains &amp
    print(f"Problem encountered escaping text: {txt}: {e}")
    import traceback; traceback.print_exc()
    txt=xml.sax.saxutils.escape(txt)
    self.parsed, attributes, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')

self.add_iter_to_buffer(attributes.get_iterator())
```

to simply:

```python
txt = xml.sax.saxutils.escape(txt)
self.parsed, attributes, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')

self.add_iter_to_buffer(attributes.get_iterator())
```

What do you think?

Reply:
It would be best if we knew the codepaths. Is it possible there's already escaped stuff there? Double-escaping would be bad.

I would only autoescape if we were sure, we never pass already escaped stuff.

Reply:
Fair point, thanks